daemon/info.go

```diff
@@ -111,3 +112,13 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// The uppercase and the lowercase are available for the proxy settings.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func getProxyEnv(key string) string {
+	proxyValue := os.Getenv(strings.ToUpper(key))
+	if proxyValue == "" {
+		return os.Getenv(key)
```

Comment:
Still relies on the case being passed to the function. If you call `getProxyEnv("FOO")`, it won't pick up the env var `foo`.